### DISCUSSION TITLE
Änderung der Akkuschonung

### DISCUSSION THREAD

**Discussion by iReckyy** [wiggal/GEN24_Ladesteuerung#84] [Ideas]
Um den Ladestrom des Akkus beim Vollwerden zu begrenzen könnte man einen Faktor in die Berechnung einfügen, was ein sinken des Strom bewirkt wenn sich der Akku 100% SOC nähert.

Vorschlag:

```
#Wenn Akkuschonung > 0 ab 80% Batterieladung mit Ladewert runter fahren
                    if Akkuschonung > 0:
                        Ladefaktor = 1
                        BattStatusProz_Grenze = 100
                    if BattStatusProz > 80:
                            Ladefaktor = 0.2
                            AkkuSchonGrund = '80%, Ladewert = 0.2C'
                            BattStatusProz_Grenze = 80
                    if BattStatusProz > 90:
                           Ladefaktor = (100 - BattStatusProz)*0.02
                           AkkuSchonGrund = '90%, Ladewert = 0.1C'
                           BattStatusProz_Grenze = 90


```

edit. ich komme mit der Formatierung nicht klar...

**Comment by wiggal** [comment]
Prinzipiell ist die Formel schon recht gut, aber die WRSchreibGrenzen, die bei der Akkuschonung zwar heruntergesetzt sind, werden ein häufiges Schreiben der Ladegrenze verhindern.
Die Schreibvorgänge möglichst gering zu halten war ursprünglich auch ein Ziel der WRSchreibGrenzen, da die Schreibzugriffe den GEN24 belasten (wie viele Schreibzugriffe möglich sind, hat mir bisher niemand sagen können).

Auch müssen wir verhindern, dass der Ladewert zu weit nach unten geht, was aber nicht das Problem wäre.

Aus diesem Grund würde ich höchsten noch eine Stufe einbauen wollen, evtl. bei 95% mit der Formel, die ich bisher bei 90% eingebaut hatte.
```
                        if BattStatusProz > 95:
                            Ladefaktor = 0.1 * Akkuschonung
                            AkkuSchonGrund = '95%, Ladewert = Ladefaktor + 'C'
                            BattStatusProz_Grenze = 95
```

Damit kann jeder für sich entscheiden, ob er bei 95% einen weiteren Schreibvorgang will. 
Bei Akkuschonung =1 bleibt der Ladewert wie bei 90%
Bei Akkuschonung =0.5 erfolgt ein Schreibvorgang mit neuem Ladewert

**Reply by iReckyy** [reply]
Das ist auch in Ordnung.

Gibt es eine maximale Anzahl an Schreibvorgängen für den GEN24 oder sollte man eine gewisse Anzahl an Schreibvorgängen pro Minute/Stunde nicht überschreiten?

**Reply by wiggal** [reply]
Wie gesagt ist nicht klar, da keiner weiß was für ein Speicher-Typ verbaut ist.
Vermutlich ein EPROM, dann hat man angeblich 100.000 Schreibvorgänge garantiert.
Wären dann 14 pro Tag 20Jahre oder so.
Also versuche ich durchschnittlich 10 Schreibzugriffe am Tag nicht zu überschreiten

**Comment by wiggal** [comment]
OK, ich habe jetzt im aktuellen Entwicklerbranch `CONFIG_DIR` mal folgendes eingebaut:

_Der Wert `Akkuschonung` in der charge.ini kann nun auch auf einen Wert zwischen 0 und 1 gesetzt werden.
  Dadurch wird der Ladewert bei einem Akkustand ab 95% um den gesetzten Faktor vermindert,
  z.B:: 0.1C = 1100W, Akkuschonung = 0.5, Dann ist der Ladewert 550W (= 0.1 x 0.5)_

**Reply by RienSte** [reply]
Funktioniert einwandfrei, allerdings muss ich dann die BattVollUm etwas nach vorne korrigieren... wenn man's genau auf 100% haben will.

Btw: habe ich gestern festgestellt, dass bei meinem HVM 11 bei 500 W Ladung das Balancing ewig braucht. Bei den vorherigen ~1 kW war das nach 30 Minuten erledigt.

vorher:
![image](https://github.com/user-attachments/assets/d120037f-79ef-448e-adcb-5ac8066d823b)

jetzt:
![image](https://github.com/user-attachments/assets/acf641b4-32ec-4f2d-bbb5-965ac6a70ef8)
Gestern wurden zB gar nie die 100% erreicht.

Macht im Grunde nichts, aber ist doch interessant zu sehen. Gibts da seitens BYD irgendeine Balancing-Logik oder so? Wisst ihr da was?

**Reply by wiggal** [reply]
Über die Balancing-Logik kann ich dir nichts sagen, was hast du den für eine Firmware auf dem HVM?
Bei dem Balancing hat sich vor allem bei der Version 3.29 einiges verändert, macht er viel häufiger.

Eine Möglichkeit wäre noch, wenn der Wert von `Akkuschonung` <= 0,5 ist `BattVollUm` nicht nur um eine Stunde vorzuziehen wie jetzt, sondern um zwei Stunden.

**Reply by RienSte** [reply]
Muss ich mal schauen, den HVM hat der Elektriker bei Inbetriebnahme auf Stand gebracht, ich war seit dem nicht mehr dran.

Eine Zusatzlogik müsste imho nicht sein, das weiß man ja eh wenn man die Akkuschonung auf 0.5 setzt. Also ich hab die BattVollUm jetzt einfach selber um eine weitere Stunde vor gerückt.
